aws/resource_aws_security_group_rule.go

```diff
@@ -689,10 +705,15 @@ func setFromIPPerm(d *schema.ResourceData, sg *ec2.SecurityGroup, rule *ec2.IpPe
 	if len(rule.UserIdGroupPairs) > 0 {
 		s := rule.UserIdGroupPairs[0]
 
+		accountPrefix := aws.StringValue(s.UserId)
```

Comment:
Now that we are setting local state from the remotely fetched rule, we should populate the local state with the full data `1234567/sg-12345`. What I realize though is with the `DiffSuppressFunc` this code becomes inert really.... we regardless of what is stored we will only diff the sg-ids... to me it feels more correct to include the information though

Reply:
This is where the flapping was caused, because previously we saved to local state only the SG ID. So for users who specified `1234567/sg-1245` in their config, TF thought the field was constantly different --> `1234567/sg-1245 != sg-1245`. Ultimately though once I added a SupressFunc as mentioned none of this particular change matters